simulator/src/main/java/com/hazelcast/simulator/worker/WorkerCommandRequestProcessor.java

```diff
@@ -355,10 +359,14 @@ private HazelcastInstance getHazelcastInstance() {
 
         private final String testId;
 
-        public CommandThread(Command command, String testId) {
+        public CommandThread(String testId, TestPhase testPhase) {
             this.testId = testId;
 
-            commands.put(testId, command);
+            TestPhase runningPhase = testPhases.putIfAbsent(testId, testPhase);
```

Comment:
This is also a change in behavior. If there is still a phase running when trying to start another one, we throw an `IllegalStateException`. Again, I think this should help to get more information about the current system state in case of a race.